pkg/safeguards/kustomize-helm-helpers.go

```diff
@@ -0,0 +1,110 @@
+package safeguards
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+	"helm.sh/helm/v3/pkg/chartutil"
+)
+
+// DirectoryType represents the type of directory.
+type DirectoryType int
+
+const (
+	Unknown   DirectoryType = iota // Default value if type cannot be determined
+	Helm                           // Represents a Helm chart directory
+	Kustomize                      // Represents a Kustomize configuration directory
+)
+
+func CreateTempDir() (string, error) {
+	log.Infof("Creating temporary directory for rendered kustomize/helm manifests")
+	tempDir, err := os.MkdirTemp("", "")
+	if err != nil {
+		log.Errorf("Error creating temporary directory to store rendered kustomize/helm manifests: %s", err)
+		return "", fmt.Errorf("failed to create temp directory: %w", err)
+	}
+
+	return tempDir, nil
+}
+
+func GetDirectoryType(dirPath string) DirectoryType {
+	if isHelm, _ := isHelmChart(dirPath); isHelm {
+		return Helm
+	}
+	// } else if isKustomizeConfig(dirPath) {
+	// 		return Kustomize
+	// }
+
+	return Unknown
+}
+
+func isHelmChart(chartDir string) (bool, error) {
```

Comment:
make it public so you can use it within validate.go

Reply:
For now, this is only going to be used in the GetDirectoryType() function above it. If there's a need to use it separately to validate a Chart.yaml passed directly in cli, I'll make it public.